crd-generator/api/src/test/java/io/fabric8/crd/generator/v1/JsonSchemaTest.java

```diff
@@ -177,6 +178,16 @@ void shouldAugmentPropertiesSchemaFromAnnotations() throws JsonProcessingExcepti
     // check ignored fields
     assertFalse(spec.containsKey("ignoredFoo"));
     assertFalse(spec.containsKey("ignoredBar"));
+
+    final JSONSchemaProps kubernetesValidation = spec.get("kubernetesValidation");
+    final List<ValidationRule> kubernetesValidationRules = kubernetesValidation.getXKubernetesValidations();
+    assertNotNull(kubernetesValidationRules);
+    assertEquals(1, kubernetesValidationRules.size());
```

Comment:
Do you mind testing the content?

Reply:
Yes, I'm not finished with this PR. 

I want to add some more tests which ensures that all parameters are passed through. 
But on my opinion we don't need to test the value content further. It's up to a developer to choose the right validation rules and the CRDGenerator should not contain additional logic to understand those rules.

Reply:
good for me, thanks :+1: